sql/api/src/main/scala/org/apache/spark/sql/catalyst/util/QuotingUtils.scala

```diff
@@ -71,4 +71,17 @@ object QuotingUtils {
 
     builder.toString()
   }
+
+  def escapeSpecialChar(str: String): String = {
+    val builder = new StringBuilder
+
+    str.foreach {
+      case '_' => builder ++= "\\_"
+      case '%' => builder ++= "\\%"
+      // TODO Support for escaping more special characters
```

Comment:
I think we should at least also escape `'`, as the generated sql is like `" LIKE '" + value + "%'";`